mullvad-daemon/src/main.rs

```diff
@@ -916,3 +945,220 @@ fn running_as_admin() -> bool {
     // TODO: Check if user is administrator correctly on Windows.
     true
 }
+
+// Windows service implementation
+
+#[cfg(windows)]
+mod system_service {
+    use super::*;
+
+    use std::ffi::OsString;
+    use std::sync::atomic::{AtomicUsize, Ordering};
+
+    use windows_service::service::{ServiceAccess, ServiceControl, ServiceControlAccept,
+                                   ServiceErrorControl, ServiceExitCode, ServiceInfo,
+                                   ServiceStartType, ServiceState, ServiceStatus, ServiceType};
+    use windows_service::service_control_handler::{self, ServiceControlHandlerResult,
+                                                   ServiceStatusHandle};
+    use windows_service::service_dispatcher;
+    use windows_service::service_manager::{ServiceManager, ServiceManagerAccess};
+
+    static SERVICE_NAME: &'static str = "MullvadVPN";
+    static SERVICE_DISPLAY_NAME: &'static str = "Mullvad VPN Service";
+
+    pub fn run() -> Result<()> {
+        let windows_directory = ::std::env::var_os("WINDIR").unwrap();
+        let global_temp_directory = PathBuf::from(windows_directory).join("Temp");
+        let main_log_file = Some(global_temp_directory.join("mullvad-daemon-service.log"));
+
+        logging::init_logger(log::LevelFilter::Debug, main_log_file.as_ref(), true)
```

Comment:
TTY logger is not really neeeded when running as a service but I didn't bother to turn it off. Ideally we should.